api/v1alpha1/etcdcluster_types.go

```diff
@@ -31,7 +43,8 @@ type EtcdClusterSpec struct {
 	// Size is the expected size of the etcd cluster.
 	Size int `json:"size"`
 	// Version is the expected version of the etcd container image.
-	Version string `json:"version"`
+	Version string         `json:"version"`
+	Tls     TLSCertificate `json:"tls,omitempty"`
 }
```

Comment:
Suggest something like below (exclude the operatorSecret),
```
type MemberSecrets struct {
	PeerSecret   string `json:"peerSecret"`
	ServerSecret string `json:"serverSecret"`
}

// TLSCertificate defines the certificate issued by the certificate provider
type TLSCertificate struct {
	Member   MemberSecrets `json:"member"`
	Provider string        `json:"provider"`
}

// EtcdClusterSpec defines the desired state of EtcdCluster.
type EtcdClusterSpec struct {
	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
	// Important: Run "make" to regenerate code after modifying this file

	// Size is the expected size of the etcd cluster.
	Size int `json:"size"`
	// Version is the expected version of the etcd container image.
	Version string          `json:"version"`
	TLS     *TLSCertificate `json:"tls,omitempty"`
}
```

Reply:
Updating the structs, thank you!